core/src/main/java/com/sequenceiq/cloudbreak/job/nodestatus/NodeStatusCheckerJob.java

```diff
@@ -86,24 +92,33 @@ protected void executeTracedJob(JobExecutionContext context) throws JobExecution
     }
 
     private void executeNodeStatusCheck(Stack stack) {
-        boolean nodeStatusCheckEnabled = true;
-        CdpNodeStatusRequest.Builder requestBuilder = CdpNodeStatusRequest.Builder.builder();
-        String accountId = Crn.safeFromString(stack.getResourceCrn()).getAccountId();
-        if (StackType.WORKLOAD.equals(stack.getType())) {
-            nodeStatusCheckEnabled = entitlementService.datahubNodestatusCheckEnabled(accountId);
-        }
-        if (nodeStatusCheckEnabled) {
-            if (StackType.WORKLOAD.equals(stack.getType())) {
-                requestBuilder.withMetering(true);
-            }
-            CdpNodeStatusRequest request = requestBuilder.withCmMonitoring(true).withSkipObjectMapping(true).build();
-            CdpNodeStatuses statuses = nodeStatusService.getNodeStatuses(stack, request);
+        getNodeStatusRequest(stack).ifPresent(nodeStatusRequest -> {
+            CdpNodeStatuses statuses = nodeStatusService.getNodeStatuses(stack, nodeStatusRequest);
             processNodeStatusReport(statuses.getNetworkReport(), NodeStatusProto.NodeStatus::getNetworkDetails, stack, "Network");
             processNodeStatusReport(statuses.getServicesReport(), NodeStatusProto.NodeStatus::getServicesDetails, stack, "Services");
-            processNodeStatusReport(statuses.getSystemMetricsReport(), NodeStatusProto.NodeStatus::getSystemMetrics,
-                    stack, "System metrics");
+            processNodeStatusReport(statuses.getSystemMetricsReport(), NodeStatusProto.NodeStatus::getSystemMetrics, stack, "System metrics");
+            processNodeStatusReport(statuses.getMeteringReport(), NodeStatusProto.NodeStatus::getMeteringDetails, stack, "Metering");
             processCmMetricsReport(statuses.getCmMetricsReport(), stack);
+        });
+    }
+
+    private Optional<CdpNodeStatusRequest> getNodeStatusRequest(Stack stack) {
+        if (StackType.WORKLOAD.equals(stack.getType()) &&
+                !entitlementService.datahubNodestatusCheckEnabled(Crn.safeFromString(stack.getResourceCrn()).getAccountId())) {
+            return Optional.empty();
+        }
+        Telemetry telemetry = componentConfigProviderService.getTelemetry(stack.getId());
+        CdpNodeStatusRequest.Builder requestBuilder = CdpNodeStatusRequest.Builder.builder();
+        if (!Objects.isNull(telemetry) && telemetry.isComputeMonitoringEnabled()) {
+            return Optional.of(requestBuilder.withNetworkOnly().build());
+        }
+        if (StackType.WORKLOAD.equals(stack.getType())) {
+            requestBuilder.withMetering();
         }
+        return Optional.of(requestBuilder
+                .withCmMonitoring()
+                .withSkipObjectMapping()
+                .build());
```

Comment:
Please put this in an else branch and similarly put the code below `return Optional.of(requestBuilder.withNetworkOnly().build());` also in an else branch. It's easier to read and understand what the function does if it's broken into if-else branches instead of returning mid-function.